### PR TITLE
Unicode awareness

### PR DESCRIPTION
**Pull request by Rohesie**
Sources:
https://github.com/tgstation/tgstation/pull/48304
https://github.com/tgstation/tgstation/pull/48512

Plus other small (few lines long) bugfixes before and after those ones, related to.

This PR will probably come back to haunt me a lot, but it's necessary.

### CODE REVIEW

File: code/_compile_options.dm [hunk 1]
```diff
@@ -39,6 +39,26 @@
 #error You need version 512 or higher
 #endif
 
+//Compatability -- These procs were added in 513.1493, not 513.1490
+//Which really shoulda bumped us up to 514 right then and there but instead Lummox is a dumb dumb
+#if DM_BUILD < 1493
```
Comment (spookydonut) [inline code/_compile_options.dm]: just stop supporting < 1493 after this PR tbh

**Comment by Rohesie** [comment]
As soon as /tg/ drops support we'll do the same.

File: code/__HELPERS/sanitize_values.dm [hunk 2]
```diff
@@ -69,29 +69,37 @@
 	return default
 
 
-/proc/sanitize_hexcolor(color, default = "#000000")
+/proc/sanitize_hexcolor(color, desired_format = 3, include_crunch = FALSE, default = "#000000")
+	var/crunch = include_crunch ? "#" : ""
 	if(!istext(color))
-		return default
+		color = ""
```
Comment (psykzz) [inline code/__HELPERS/sanitize_values.dm]:
```suggestion
		return default 
```

Its only going to fail later.

**Comment by Rohesie** [comment]
This PR is ready for merge.

**Pull request merged**